libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -852,11 +852,7 @@ - (void)setupWithCredentials:(SFOAuthCredentials*) credentials {
 
 - (void)restartAuthentication {
     @synchronized (self.authBlockList) {
-        if (!self.authenticating) {
-            [self log:SFLogLevelWarning format:@"%@: Authentication manager is not currently authenticating.  No action taken.", NSStringFromSelector(_cmd)];
-            return;
-        }
-        [self log:SFLogLevelInfo format:@"%@: Restarting in-progress authentication process.", NSStringFromSelector(_cmd)];
+        [self log:SFLogLevelInfo format:@"%@: Restarting authentication process.", NSStringFromSelector(_cmd)];
```

Comment:
restartAuthentication is new in 5.2, and is called only from loginViewController:didChangeLoginHost:. That happens when you are changing host during a normal login flow.

I wanted to call it also from hostListViewController:didChangeLoginHost:. That one happens when you are changing flow after cancelling out of an advanced login flow.

In the first case authenticating is happening but not in the second case.

Either I change restartAuthentication to always start an authentication if none is happening.
Or I could introduce a new method startAuthentication and call it in the second case.